### PR TITLE
Add flag to test fast jsx

### PR DESCRIPTION
**Pull request by jackpope**
Following #28768, add a path to testing Fast JSX on www.

We want to measure the impact of Fast JSX and enable a path to testing before string refs are completely removed in www (which is a work in progress). 

Without `disableStringRefs`, we need to copy any object with a `ref` key so we can pass it through `coerceStringRef()` and copy it into the object. This de-opt path is what is gated behind `enableFastJSXWithStringRefs`.

The additional checks should have no perf impact in OSS as the flags remain true there and the build output is not changed. For www, I've benchmarked the addition of the boolean checks with values cached at module scope. There is no significant change observed from our benchmarks and any latency will apply to test and control branches evenly. This added experiment complexity is temporary. We should be able to clean it up, along with the flag checks for `enableRefAsProp` and `disableStringRefs` shortly.

### CODE REVIEW

**Comment by sebmarkbage** [comment]
Dynamic flags in JSX is a bad idea since it'll slow down the default case. You can maybe try it for bugs but you'll slow down both the test and control so perf testing isn't good and also you slow down everyone.

**Comment by kassens** [comment]
Is 1 additional boolean check really that critical here? It seems like compared to all the allocations going on that would be minimal?

**Comment by yungsters** [comment]
At some point, we're going to start rolling out each of the feature flags affecting `jsx-runtime`, such as `enableRefAsProp` and `disableStringRefs`. When that happens, these flags will have to become dynamic for at least a short period of time for stabilization in production.

I think there's consensus that they'll introduce non-zero overhead, but maybe there's disagreement about how much and the magnitude of regression. As long as we're not regressing open source bundles, I think that the regression can be a calculated risk for Meta-specific experimentation.

For what it's worth, my expectation is that as long as both the control and test groups have the additional overhead, our experimentation results should still be sufficiently meaningful. Knowing that these code paths are hot, we can also minimize the time that the feature flag is configured to be dynamic.

In conclusion, I don't think we need to block on forking the entire build artifact before trying this out in production.

That all said, I *would* prefer to see an `enableFastJSX` feature flag instead of an `enableFastJSXWithStringRefs` feature flag.

**Comment by jackpope** [comment]
@yungsters My intention here was to both have a path to testing this sooner that we expect the two blocking flags to roll out internally, and also to generally create a stable experiment to understand the impact of fast jsx. The built in assumption in the initial implementation was that we had a while before the blocking flags were enabled but I agree its better to be explicit, especially since the story there might be different on the RN side. Plus now we have the string ref codemod.

My next steps here will be:
- Update this implementation to have a more general `enableFastJSX` flag like you suggest. On in OSS, dynamic for our tests
- Run a benchmark with the extra boolean checks so we have an understanding (for this and for the future) of what kind of overhead will occur from the check
- Clean up whatever tests are failing here and get this in a landable state so we can start testing assuming said benchmark doesn't majorly regress control

**Review by yungsters — APPROVED** [review]
